src/Model/FileExplorer.re

```diff
@@ -8,7 +8,8 @@ type t = {
   isOpen: bool,
   scrollOffset: [ | `Start(float) | `Middle(float) | `Reveal(int)],
   active: option(string), // path
-  focus: option(string) // path
+  focus: option(string), // path
+  decorations: StringMap.t(list(SCMDecoration.t)),
```

Comment:
decorations are stored directly on the `fileExplorer` in a `StringMap` that maps paths to decorations instead of on the nodes themselves, since the nodes might not have been loaded yet! It's also easier to update them when you don't have to search a tree that may or may not be there etc.

Reply:
Thanks for the explanation! Makes sense 👍